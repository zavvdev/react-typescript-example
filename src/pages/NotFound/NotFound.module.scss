@import "src/styles/theme";

.root {
  text-align: center;
  padding-top: 142px;

  span {
    font-size: 30px;
    color: $cBlackLight;
  }

  .btn {
    margin: 42px auto 0 auto;
  }
}