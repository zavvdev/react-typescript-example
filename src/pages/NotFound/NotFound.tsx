import { useNavigate } from 'react-router-dom';
import { GENERAL_ROUTES } from 'src/constants/routes';
import NotFoundView from 'src/pages/NotFound/NotFound.view';

const NotFound = () => {
  const navigate = useNavigate();

  const handleHomeClick = () => {
    navigate(GENERAL_ROUTES.home, {
      replace: true,
    });
  };

  return (
    <NotFoundView
      handleHomeClick={handleHomeClick}
    />
  );
};

export default NotFound;
