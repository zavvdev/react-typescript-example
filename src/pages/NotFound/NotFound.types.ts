export interface NotFoundViewProps {
  handleHomeClick: () => void;
}
