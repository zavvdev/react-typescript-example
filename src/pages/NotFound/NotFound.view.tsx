import { useTranslation } from 'react-i18next';
import Button from 'src/components/Button/Button';
import Namespace from 'src/i18n/namespace';
import classes from 'src/pages/NotFound/NotFound.module.scss';
import { NotFoundViewProps } from 'src/pages/NotFound/NotFound.types';

const NotFoundView = (props: NotFoundViewProps) => {
  const { handleHomeClick } = props;
  const { t } = useTranslation(Namespace.NotFound);

  return (
    <div className={classes.root}>
      <span>{t('code')}</span>
      <h1>{t('text')}</h1>
      <Button className={classes.btn} onClick={handleHomeClick}>
        {t('buttons.home')}
      </Button>
    </div>
  );
};

export default NotFoundView;
