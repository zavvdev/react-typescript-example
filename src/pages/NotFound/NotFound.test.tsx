import { screen, fireEvent } from '@testing-library/react';
import render from 'src/tests/utils/render';
import { mockNavigate } from 'src/tests/mocks/node-modules/react-router-dom';
import { GENERAL_ROUTES } from 'src/constants/routes';
import NotFound from 'src/pages/NotFound/NotFound';

describe('<NotFound />', () => {
  test('should render NotFound page (snapshot)', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have http code', () => {
    render(<NotFound />);
    expect(screen.getByText('code')).toBeInTheDocument();
  });

  test('should have a text', () => {
    render(<NotFound />);
    expect(screen.getByText('text')).toBeInTheDocument();
  });

  test('should have a Home button', () => {
    render(<NotFound />);
    const btn = screen.getByText('buttons.home');
    expect(btn.getAttribute('type')).toBe('button');
  });

  test('should call navigate method', () => {
    render(<NotFound />);
    const btn = screen.getByText('buttons.home');
    fireEvent.click(btn);
    expect(mockNavigate).toBeCalledWith(
      GENERAL_ROUTES.home,
      { replace: true },
    );
  });
});
