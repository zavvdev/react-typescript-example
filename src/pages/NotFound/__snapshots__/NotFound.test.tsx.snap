// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render NotFound page (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <span>
      code
    </span>
    <h1>
      text
    </h1>
    <button
      class="root btn"
      type="button"
    >
      buttons.home
    </button>
  </div>
</DocumentFragment>
`;
